### PR TITLE
Realistic static B and it's initialization

### LINKED ISSUE #93: SOCA: Add training step for BUMP
Related to this [issue](https://github.com/noaa-emc/godas/issues/342)

### PR DESCRIPTION
**Pull request by guillaumevernieres**
## Description
Addition of all the necessary steps to initialize `bump_nicas` sequentially:
- generate scales using `soca_setcorscales.x`
- compute the correlation operator with bump_nicas based on the scales above using `soca_error_covariance_training.x`
- replaced "identity C" by bump_nicas C in the 3dvar fgat
checks a box in the [godas issue#342](https://github.com/noaa-emc/godas/issues/342)
### Issues addresse
- fixes #93

### CODE REVIEW

**Comment by guillaumevernieres** [comment]
Note to self: not finding `ncks` on ~Hera~ Orion ~~despite `nco/4.9.1` being loaded ...~~

File: parm/soca/berror/soca_setcorscales.yaml [hunk 1]
```diff
@@ -0,0 +1,44 @@
+resolution:
+  mom6_input_nml: mom_input.nml
+  fields metadata: ./fields_metadata.yaml
+
+date: 0001-01-01T00:00:00Z
```
Comment (CoryMartin-NOAA) [inline parm/soca/berror/soca_setcorscales.yaml:5]: If this runs every cycle, why not set the date appropriately here (and in the previous file?)

File: scripts/exgdas_global_marine_analysis_prep.py [hunk 2]
```diff
@@ -112,10 +123,45 @@ def gen_bkg_list(bkg_path='.', file_type='MOM', yaml_name='bkg.yaml'):
...
+                              'berror',
+                              'soca_setcorscales.yaml')
+ufsda.disk_utils.symlink(corscales_yaml,
+                         os.path.join(stage_cfg['stage_dir'], 'soca_setcorscales.yaml'))
+
+# generate yaml for bump C
+# TODO (Guillaume): move the possible vars somewhere else
+vars3d = ['tocn', 'socn', 'uocn', 'vocn', 'chl', 'biop']
+vars2d = ['ssh', 'cicen', 'hicen', 'hsnon',
+          'swh',
```
Comment (CoryMartin-NOAA) [inline scripts/exgdas_global_marine_analysis_prep.py]: is there a particular reason why this one var is alone on this line?

**Review by CoryMartin-NOAA — COMMENTED** [review]
couple of q's/comments but mostly looks good

File: parm/soca/berror/soca_setcorscales.yaml (hunk 1 above)
Comment (guillaumevernieres) [reply parm/soca/berror/soca_setcorscales.yaml:5]: It's done once, just not reflected yet in the scripts. I'll add switches in the future.

File: scripts/exgdas_global_marine_analysis_prep.py (hunk 2 above)
Comment (guillaumevernieres) [reply scripts/exgdas_global_marine_analysis_prep.py]: I was going to say I organized them by domains: ocean, sea-ice. wave and atm-fluxes ... But clearly I didn't.
How about this:
```python
vars3d = ['tocn', 'socn', 'uocn', 'vocn', 
          'chl', 'biop']
vars2d = ['ssh',
          'cicen', 'hicen', 'hsnon',
          'swh',
          'sw', 'lw', 'lw_rad', 'lhf', 'shf', 'us']
```
or I could just move `swh` to L141 ...

Comment (CoryMartin-NOAA) [reply scripts/exgdas_global_marine_analysis_prep.py]: I would just move swh, seems a bit excessive to sort them for this purpose, but I defer to you

**Review by CoryMartin-NOAA — APPROVED** [review]
Looks good to me

**Pull request merged**